contracts/liquid-collateral/src/contract.rs

```diff
@@ -0,0 +1,1089 @@
+        ExecuteMsg::EndRoundBid(msg) => end_round_bid(deps, env, info, msg),
+        ExecuteMsg::WithdrawBid => withdraw_bid(deps, env, info),
+        ExecuteMsg::ResolveAuction => resolve_auction(deps, env, info),
+    }
+}
+
+#[entry_point]
+pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        // Handle the GetState query
+        QueryMsg::State {} => to_json_binary(&query_get_state(deps)?),
+        QueryMsg::Bid { address } => to_json_binary(&query_bid(deps, address)?),
+        QueryMsg::Bids {} => to_json_binary(&query_get_bids(deps)?),
+        QueryMsg::SortedBids {} => to_json_binary(&query_sorted_bids(deps)?),
+    }
+}
+/*
+
+## Create position with reply
+ - based on the passed position arguments, method is sending MsgCreatePosition to the cl module on Osmosis
+ - in the reply contract is updating the state with the position information.
+*/
+pub fn create_position(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: CreatePositionMsg,
+) -> Result<Response, ContractError> {
+    let mut state = STATE.load(deps.storage)?;
+
+    // Check if the position_id already exists
+    if state.position_id.is_some() {
+        return Err(ContractError::PositionAlreadyExists {});
+    }
+
+    // Only owner can create position - if present
+    match &state.project_owner {
+        Some(owner) if info.sender != *owner => {
+            return Err(ContractError::Unauthorized {});
+        }
+        _ => {}
+    }
+
+    /*
+    The order of the tokens which will be put in tokens provided is very important
+    on osmosis they check lexicographical order: https://github.com/osmosis-labs/osmosis/blob/main/x/concentrated-liquidity/types/msgs.go#L42C2-L44C3
+    ibc/token should go before uosmo token for example
+    if order is not correct - tx will fail!
+     */
+
+    let mut tokens_provided: Vec<OsmosisCoin> = Vec::new();
+    let mut token_min_amount0 = "0".to_string();
+    let mut token_min_amount1 = "0".to_string();
+
+    for coin in &info.funds {
+        let osmo_coin = OsmosisCoin {
+            denom: coin.denom.clone(),
+            amount: coin.amount.to_string(),
+        };
+
+        if coin.denom == state.principal_denom {
+            if state.principal_first {
+                token_min_amount0 = msg.principal_token_min_amount.to_string();
+            } else {
+                token_min_amount1 = msg.principal_token_min_amount.to_string();
+            }
+        } else if coin.denom == state.counterparty_denom {
+            if state.principal_first {
+                token_min_amount1 = msg.counterparty_token_min_amount.to_string();
+            } else {
+                token_min_amount0 = msg.counterparty_token_min_amount.to_string();
+            }
+        } else {
+            return Err(ContractError::AssetNotFound {});
+        }
+
+        tokens_provided.push(osmo_coin);
+    }
+
+    // Create position message
```

Comment:
this is the type of comment that doesn't make sense (I guess it's just a remainder of sketching the code out) since it just says what the line after it states very plainly, so it doesn't add any value